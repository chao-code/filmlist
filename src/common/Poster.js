import PropTypes from 'prop-types'
import cropPoster from '../utils/cropPoster'

const Poster = ({ poster, crop }) => {
  const bsClass = 'img-fluid card'
  const src = crop ? cropPoster(poster) : poster


  return (
    <img className={bsClass} src={src} alt="Poster" />
  )
}

Poster.propTypes = {
  poster: PropTypes.string,
  crop: PropTypes.bool,
}

Poster.defaultProps = {
  poster: '',
  crop: false,
}

export default Poster
