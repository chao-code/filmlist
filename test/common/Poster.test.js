import { shallow } from 'enzyme'
import Poster from '../../src/common/Poster'

jest.mock('../../src/utils/cropPoster', () => jest.fn(() => 'cropped'))

describe('<Poster />', () => {
  it('should render correctly', () => {
    const poster = 'poster.jpg'
    const wrapper = shallow(<Poster poster={poster} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should crop the poster', () => {
    const poster = 'poster.jpg'
    const wrapper = shallow(<Poster poster={poster} crop />)
    const src = wrapper.find('img').prop('src')
    expect(src).toBe('cropped')
  })
})
