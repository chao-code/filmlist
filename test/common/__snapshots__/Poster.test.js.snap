// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Poster /> should render correctly 1`] = `
<img
  alt="Poster"
  className="img-fluid card"
  src="poster.jpg"
/>
`;
